hw2/03-count.js

```diff
@@ -0,0 +1,26 @@
+// Add your code here
+const wordInput = document.getElementById("wordInput");
+const paragraph = document.getElementById("paragraph");
+
+wordInput.addEventListener("keydown", (event) => {
+  const originText = paragraph.textContent;
+  if (event.key === "Enter") {
+    const wordToHighlight = event.target.value.trim();
+
+    if (wordToHighlight.length > 0) {
+      //Using flag "gi" to capture all matched pattern text within a string instead of just the first matched.
+      const regex = new RegExp(wordToHighlight, "gi");
+
+      //Using $& to inserts the matched substring that was defined from RegExp()
+      const matchedText = paragraph.innerHTML.replace(regex, "<mark>$&</mark>");
```

Comment:
small nit: instead of using `<mark` you can use a `<text>` tag and give it a css attribute, or have an attached stylesheet and use an id attribute to style the highlighted word.  Using mark can add spacing, while just styling with a color background of the text, will simply highlight.  Hope thats helpful!

Reply:
likewise, if you enter a word, the old ones stay highlighted, but I see this already been addressed in miles' review.